File: app/src/main/java/com/smile/watchmovie/fragment/PersonFragment.java
package com.smile.watchmovie.fragment;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.google.android.gms.auth.api.signin.GoogleSignIn;
import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
import com.smile.watchmovie.eventBus.EventNotifyLogIn;
import com.smile.watchmovie.activity.ChoosePaymentActivity;
import com.smile.watchmovie.activity.FavoriteFilmActivity;
import com.smile.watchmovie.activity.HistoryWatchFilmActivity;
import com.smile.watchmovie.activity.InfoAccountActivity;
import com.smile.watchmovie.activity.LoginActivity;
import com.smile.watchmovie.R;
import com.smile.watchmovie.activity.PrivateSettingActivity;
import com.smile.watchmovie.api.ApiService;
import com.smile.watchmovie.databinding.FragmentPersonBinding;
import com.smile.watchmovie.model.WeatherResponse;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PersonFragment extends Fragment {
    private FragmentPersonBinding binding;
    String nameUser, idUser, is_vip;
    private SharedPreferences sharedPreferences;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentPersonBinding.inflate(inflater, container, false);
        sharedPreferences = requireActivity().getSharedPreferences("MyPrefs", Context.MODE_PRIVATE);
        getInfoUserInDB();
        setupInfo();
        onClickItem();
        showWeather();

        return binding.getRoot();
    }

    private void setupInfo() {
        if (!is_vip.equals("0")) {
            binding.ivVip.setVisibility(View.VISIBLE);
            binding.tvTitlePayDetail.setText(R.string.account_vip);
        } else {
            binding.ivVip.setVisibility(View.GONE);
            binding.tvTitlePayDetail.setText(R.string.buy_title);
        }

        if (!nameUser.equals("")) {
            binding.tvNameAccount.setText(nameUser);
            binding.ivAvtAccount.setVisibility(View.VISIBLE);
            loadAvatarUser();
        } else {
            binding.ivAvtAccount.setVisibility(View.GONE);
            binding.tvNameAccount.setText(requireActivity().getString(R.string.login));
        }
    }

    private void onClickItem() {
        binding.tvHistory.setOnClickListener(v -> {
            if (idUser == null || idUser.equals("")) {
                Toast.makeText(requireActivity(), getString(R.string.not_logged_in_message), Toast.LENGTH_SHORT).show();
            } else {
                requireActivity().startActivity(new Intent(requireActivity(), HistoryWatchFilmActivity.class));
            }
        });

        binding.tvDownload.setOnClickListener(v ->
                Toast.makeText(requireActivity(), getString(R.string.feature_deploying), Toast.LENGTH_SHORT).show()
        );

        binding.tvFavorite.setOnClickListener(v -> {
            if (idUser == null || idUser.equals("")) {
                Toast.makeText(requireActivity(), getString(R.string.not_logged_in_message), Toast.LENGTH_SHORT).show();
            } else {
                requireActivity().startActivity(new Intent(requireActivity(), FavoriteFilmActivity.class));
            }
        });

        binding.tvSetting.setOnClickListener(v ->
                requireActivity().startActivity(new Intent(requireActivity(), PrivateSettingActivity.class))
        );

        binding.tvHelpFeedback.setOnClickListener(v ->
                Toast.makeText(requireActivity(), getString(R.string.feature_deploying), Toast.LENGTH_SHORT).show()
        );

        binding.loutPay.setOnClickListener(v -> {
            if (idUser == null || idUser.equals("")) {
                Toast.makeText(requireActivity(), getString(R.string.not_logged_in_message), Toast.LENGTH_SHORT).show();
            } else {
                requireActivity().startActivity(new Intent(requireActivity(), ChoosePaymentActivity.class));
            }
        });

        binding.loutAccount.setOnClickListener(v -> {
            if (idUser.equals("")) {
                startActivity(new Intent(requireActivity(), LoginActivity.class));
            } else {
                startActivity(new Intent(requireActivity(), InfoAccountActivity.class));
            }
        });
    }

    public void showWeather() {
        ApiService.apiWeather.getWeather("Hanoi", "metric", requireActivity().getString(R.string.appId)).enqueue(new Callback<WeatherResponse>() {
            @Override
            public void onResponse(@NonNull Call<WeatherResponse> call, @NonNull Response<WeatherResponse> response) {
                WeatherResponse weatherResponse = response.body();
                if (weatherResponse != null) {
                    binding.tvWeather.setText(weatherResponse.getMain().getTemp() + "°C" + " tại Hà Nội");
                    binding.tvWeatherDetail.setText(weatherResponse.getWeather().get(0).getMain());
                }
            }

            @Override
            public void onFailure(@NonNull Call<WeatherResponse> call, @NonNull Throwable t) {
            }
        });
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onEvent(EventNotifyLogIn isLogIn) {
        getInfoUserInDB();
        setupInfo();
    }

    private void getInfoUserInDB() {
        idUser = sharedPreferences.getString("idUser", "");
        nameUser = sharedPreferences.getString("name", "");
        is_vip = sharedPreferences.getString("isVip", "0");
    }

    private void loadAvatarUser() {
        GoogleSignInAccount signInAccount = GoogleSignIn.getLastSignedInAccount(requireActivity());
        if (signInAccount != null) {
            Glide.with(this).load(signInAccount.getPhotoUrl()).into(binding.ivAvtAccount);
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (!EventBus.getDefault().isRegistered(this)) {
            EventBus.getDefault().register(this);
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        EventBus.getDefault().unregister(this);
    }
}

File: app/src/main/java/com/smile/watchmovie/fragment/PersonFragment.kt
package com.smile.watchmovie.fragment

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.bumptech.glide.Glide
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.smile.watchmovie.R
import com.smile.watchmovie.activity.BuyPremiumActivity
import com.smile.watchmovie.activity.FavoriteFilmActivity
import com.smile.watchmovie.activity.HistoryWatchFilmActivity
import com.smile.watchmovie.activity.InfoAccountActivity
import com.smile.watchmovie.activity.LoginActivity
import com.smile.watchmovie.activity.PrivateSettingActivity
import com.smile.watchmovie.api.ApiHelper
import com.smile.watchmovie.api.RetrofitBuilder
import com.smile.watchmovie.base.ViewModelFactory
import com.smile.watchmovie.databinding.FragmentPersonBinding
import com.smile.watchmovie.eventbus.EventNotifyLogIn
import com.smile.watchmovie.utils.Constant
import com.smile.watchmovie.viewmodel.PersonFragmentViewModel
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class PersonFragment : Fragment() {
    private lateinit var binding: FragmentPersonBinding
    private lateinit var sharedPreferences: SharedPreferences
    private lateinit var viewModel: PersonFragmentViewModel
    private var nameUser: String? = null
    private var idUser: String? = null
    private var isVip: String? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        /* Khởi tạo dữ liệu */
        binding = FragmentPersonBinding.inflate(inflater, container, false)
        sharedPreferences = requireActivity().getSharedPreferences(
            Constant.NAME_DATABASE_SHARED_PREFERENCES,
            Context.MODE_PRIVATE
        )
        viewModel = ViewModelProvider(
            this,
            ViewModelFactory(ApiHelper(RetrofitBuilder.apiServiceWeather))
        )[PersonFragmentViewModel::class.java]

        getInfoUserInDB()
        getDataFromAPI()
        setupInfo()
        onClickItem()

        return binding.root
    }

    private fun getDataFromAPI() {
        viewModel.getInfoWeather("Hanoi", "metric", Constant.Api.APP_ID)
            .observe(requireActivity()) {
                it?.let {
                    binding.apply {
                        tvWeather.text =
                            requireActivity().getString(R.string.title_weather, it.main.temp)
                        tvWeatherDetail.text = it.weather[0].main
                    }
                }
            }
    }

    private fun setupInfo() {
        if (isVip != "0") {
            binding.ivVip.visibility = View.VISIBLE
            binding.tvTitlePayDetail.setText(R.string.account_vip)
        } else {
            binding.ivVip.visibility = View.GONE
            binding.tvTitlePayDetail.setText(R.string.buy_title)
        }

        if (nameUser != "") {
            binding.tvNameAccount.text = nameUser
            binding.ivAvtAccount.visibility = View.VISIBLE
            loadAvatarUser()
        } else {
            binding.ivAvtAccount.visibility = View.GONE
            binding.tvNameAccount.text = requireActivity().getString(R.string.login)
        }
    }

    private fun onClickItem() {
        binding.tvHistory.setOnClickListener {
            if (idUser == null || idUser == "") {
                Toast.makeText(
                    requireActivity(),
                    getString(R.string.not_logged_in_message),
                    Toast.LENGTH_SHORT
                ).show()
            } else {
                requireActivity().startActivity(
                    Intent(
                        requireActivity(),
                        HistoryWatchFilmActivity::class.java
                    )
                )
            }
        }

        binding.tvDownload.setOnClickListener {
            Toast.makeText(
                requireActivity(),
                getString(R.string.feature_deploying),
                Toast.LENGTH_SHORT
            ).show()
        }

        binding.tvFavorite.setOnClickListener {
            if (idUser == null || idUser == "") {
                Toast.makeText(
                    requireActivity(),
                    getString(R.string.not_logged_in_message),
                    Toast.LENGTH_SHORT
                ).show()
            } else {
                requireActivity().startActivity(
                    Intent(
                        requireActivity(),
                        FavoriteFilmActivity::class.java
                    )
                )
            }
        }

        binding.tvSetting.setOnClickListener {
            requireActivity().startActivity(
                Intent(
                    requireActivity(),
                    PrivateSettingActivity::class.java
                )
            )
        }

        binding.tvHelpFeedback.setOnClickListener {
            Toast.makeText(
                requireActivity(),
                getString(R.string.feature_deploying),
                Toast.LENGTH_SHORT
            ).show()
        }

        binding.loutPay.setOnClickListener {
            if (idUser == null || idUser == "") {
                Toast.makeText(
                    requireActivity(),
                    getString(R.string.not_logged_in_message),
                    Toast.LENGTH_SHORT
                ).show()
            } else {
                requireActivity().startActivity(
                    Intent(requireActivity(), BuyPremiumActivity::class.java)
                )
            }
        }

        binding.loutAccount.setOnClickListener {
            if (idUser == "") {
                startActivity(Intent(requireActivity(), LoginActivity::class.java))
            } else {
                startActivity(Intent(requireActivity(), InfoAccountActivity::class.java))
            }
        }
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onEvent(isLogIn: EventNotifyLogIn?) {
        getInfoUserInDB()
        setupInfo()
    }

    private fun getInfoUserInDB() {
        idUser = sharedPreferences.getString(Constant.ID_USER, "")
        nameUser = sharedPreferences.getString(Constant.NAME_USER, "")
        isVip = sharedPreferences.getString(Constant.IS_VIP, "0")
    }

    private fun loadAvatarUser() {
        val signInAccount = GoogleSignIn.getLastSignedInAccount(requireActivity())
        if (signInAccount != null) {
            Glide.with(this).load(signInAccount.photoUrl).into(binding.ivAvtAccount)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (!EventBus.getDefault().isRegistered(this)) {
            EventBus.getDefault().register(this)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        EventBus.getDefault().unregister(this)
    }
}